boa/src/environment/declarative_environment_record.rs

```diff
@@ -107,27 +122,35 @@ impl EnvironmentRecordTrait for DeclarativeEnvironmentRecord {
         }
         if record.value.is_none() {
             // TODO: change this when error handling comes into play
-            panic!("Reference Error: Cannot set mutable binding for {}", name);
+            return Err(ErrorKind::ReferenceError(format!(
+                "Reference Error: Cannot set mutable binding for {}",
+                name
+            )));
         }
 
         if record.mutable {
             record.value = Some(value);
         } else if strict {
             // TODO: change this when error handling comes into play
-            panic!("TypeError: Cannot mutate an immutable binding {}", name);
+            return Err(ErrorKind::TypeError(format!(
+                "TypeError: Cannot mutate an immutable binding {}",
+                name
+            )));
         }
+
+        Ok(())
     }
 
-    fn get_binding_value(&self, name: &str, _strict: bool) -> Value {
+    fn get_binding_value(&self, name: &str, _strict: bool) -> Result<Value, ErrorKind> {
         if let Some(binding) = self.env_rec.get(name) {
-            binding
-                .value
-                .as_ref()
-                .expect("Could not get record as reference")
-                .clone()
+            if let Some(ref val) = binding.value {
+                Ok(val.clone())
+            } else {
+                Err(ErrorKind::ReferenceError("".to_string()))
+            }
         } else {
             // TODO: change this when error handling comes into play
-            panic!("ReferenceError: Cannot get binding value for {}", name);
+            panic!("Cannot get binding value for {}", name);
```

Comment:
Why is this still a panic?

Reply:
The [spec](https://tc39.es/ecma262/#sec-declarative-environment-records-getbindingvalue-n-s) calls assert (env_rec contains the binding).

Reply:
> The [spec](https://tc39.es/ecma262/#sec-declarative-environment-records-getbindingvalue-n-s) calls `assert(env_rec contains the binding)`.

Hmmm. In that case leave it as a `panic!`